model/src/main/kotlin/utils/FindingsMatcher.kt

```diff
@@ -274,5 +280,67 @@ fun associateLicensesWithExceptions(
         exception.copy(license = "${SpdxConstants.NOASSERTION} ${SpdxExpression.WITH} ${exception.license}".toSpdx())
     }
 
-    return fixedLicenses
+    return fixedLicenses.mapTo(mutableSetOf()) { it.copy(license = associateLicensesWithExceptions(it.license)) }
+}
+
+/**
+ * Process [license] for stand-alone license exceptions as part of compound expressions and associate them with
+ * applicable licenses. Orphan license exceptions will get associated by [SpdxConstants.NOASSERTION]. Return a new
+ * expression that does not contain stand-alone license exceptions anymore.
+ */
+internal fun associateLicensesWithExceptions(license: SpdxExpression): SpdxExpression {
```

Comment:
out of scope: Can we move these two "associate*" functions to `ScanResultModelMapper` as these are unrelated to the other code in findings matcher but strongly related to `toSummary()` ?

Reply:
Actually, these functions are deliberately implemented in a scanner-agnostic way, and I'm using them already in code that is not using the `ScanCodeResultModelMapper`, so I'd prefer to leave them here.